### PR TITLE
azureclient v2: add recorder for v2 client ut cases

### PR DESCRIPTION
**Pull request by MartinForReal**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
add recorder for v2 client ut cases
#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/v2/recording/recorder.go [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recording
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
```
Comment (lzhecheng) [inline pkg/azureclients/v2/recording/recorder.go:22]: nic: seperate go libs and others.

File: pkg/azureclients/v2/recording/transport.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package recording
+
+import (
+	"fmt"
+	"github.com/google/go-cmp/cmp"
```
Comment (lzhecheng) [inline pkg/azureclients/v2/recording/transport.go:21]: ditto

File: pkg/azureclients/v2/recording/recorder.go [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+	// check body as well as URL/Method (copied from go-vcr documentation)
+	rec.SetMatcher(func(r *http.Request, i cassette.Request) bool {
+		if !cassette.DefaultMatcher(r, i) {
+			return false
+		}
+
+		//// verify custom request count header (see counting_roundtripper.go)
+		//if r.Header.Get(countHeader) != i.Headers.Get(countHeader) {
+		//	return false
+		//}
```
Comment (lzhecheng) [inline pkg/azureclients/v2/recording/recorder.go:139]: forget to remove these?

File: pkg/azureclients/v2/recording/recorder.go [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+	if rec.Mode() == recorder.ModeRecording ||
+		rec.Mode() == recorder.ModeDisabled {
+		tokenCredential, err = azidentity.NewDefaultAzureCredential(nil)
+		if err != nil {
+			return nil, err
+		}
+
+		subscriptionID = os.Getenv("AZURE_SUBSCRIPTION_ID")
+		if subscriptionID == "" {
+			return nil, fmt.Errorf("required environment variable %q was not supplied", "AZURE_SUBSCRIPTION_ID")
```
Comment (lzhecheng) [inline pkg/azureclients/v2/recording/recorder.go:121]: Why not just a simple string?

**Review by lzhecheng — COMMENTED** [review]
some nits

File: pkg/azureclients/v2/recording/recorder.go (hunk 4 above)
Comment (MartinForReal) [reply pkg/azureclients/v2/recording/recorder.go]: updated. Thanks!

File: pkg/azureclients/v2/recording/recorder.go (hunk 3 above)
Comment (MartinForReal) [reply pkg/azureclients/v2/recording/recorder.go]: updated. Thanks!

File: pkg/azureclients/v2/recording/transport.go (hunk 2 above)
Comment (MartinForReal) [reply pkg/azureclients/v2/recording/transport.go]: updated

**Pull request merged**